### ISSUE TITLE
Reliable Reconnection

### ISSUE THREAD

**Issue by lgrahl** [threema-ch/threema-web#802]
This is an umbrella issue that tracks known issues related to reconnection attempts which need resolving before we can release the reconnect overhaul code introduced by #792. The rationale here is that the multiple pushes trigger race conditions in both the iOS and the Android app (which is good because at least we can now reproduce them better).

This issue is not intended for discussion. If you encounter further issues you suspect are not listed here or you have new insights, please file a new issue.

### Threema Web

* [ ] #799: Investigate why it causes these UI issues

### SaltyRTC Client JS

* [ ] https://github.com/saltyrtc/saltyrtc-client-js/issues/114

### Android App

* [ ] Must not establish multiple SaltyRTC connections of the same session (internal: ANDR-909).
* [ ] Must not wait until a previous SaltyRTC connection has been stopped / a timer fired when the affiliation id of the push (`wca`) is different. It should discard the connection and restart the web client session immediately (ANDR-908).
* [ ] Must ignore pushes with the same affiliation id (`wca`) when it is already connected or currently establishing a connection with that id (internal: ANDR-909).
* [ ] Must not ignore pending wakeups (internal: ANDR-914).

### iOS App

* [ ] Must ignore pushes with the same affiliation id (`wca`) when it is already connected or currently establishing a connection with that id (internal: IOS-751).

**Comment by lgrahl** [comment]
Just wanted to let everyone know that we've resolved a couple of race conditions in the Android app that resulted in reconnect attempts being ignored. Should be in the next release.

**Comment by ovalseven8** [comment]
- [ ] Revert 86ed2f93456845e3992df79672aa03f24796253b
